spec/std/yaml/any_spec.cr

```diff
@@ -94,11 +94,15 @@ describe YAML::Any do
       YAML.parse("").as_nil.should be_nil
     end
 
-    it "gets bool" do
+    it "gets bool", focus: true do
       YAML.parse("true").as_bool.should be_true
       YAML.parse("false").as_bool.should be_false
       YAML.parse("true").as_bool?.should be_true
       YAML.parse("false").as_bool?.should be_false
+      YAML.parse("yes").as_bool?.should be_true
+      YAML.parse("no").as_bool?.should be_false
+      YAML.parse("'yes'").as_bool?.should be_nil
+      YAML.parse("'no'").as_bool?.should be_nil
```

Comment:
These specs pass even without this PR. Instead this PR seems to only affect users of `YAML::Serializable::Unmapped`:

```crystal
require "yaml"

class Foo
  include YAML::Serializable
  include YAML::Serializable::Unmapped
end

# before
Foo.from_yaml("x: yes")      # => #<Foo:0x13732ca0ce0 @yaml_unmapped={"x" => true}>
Foo.from_yaml("x: 'yes'")    # => #<Foo:0x13732ca0bc0 @yaml_unmapped={"x" => true}>
YAML.parse("'yes'").as_bool? # => nil

# after
Foo.from_yaml("x: yes")   # => #<Foo:0x24f0b1d0cc0 @yaml_unmapped={"x" => true}>
Foo.from_yaml("x: 'yes'") # => #<Foo:0x24f0b1d0ba0 @yaml_unmapped={"x" => "yes"}>
YAML.parse("'yes'").as_bool? # => nil
```

Reply:
Ah `YAML.parse` and `YAML::Any.from_yaml` seem to go through different codepaths?

I can't see any specs specifically for `YAML::Any.from_yaml`—should I add some in this file?

And yeah this will change the parsing of `YAML::Unmapped` or if you have a field of type `YAML::Any` (which is what I was doing).

So:

```
require "yaml"

# Before
YAML::Any.from_yaml "'yes'" # => true
YAML::Any.from_yaml "yes" # => true
YAML.parse "'yes'" # => "yes"
YAML.parse "yes" # => true

# Before
YAML::Any.from_yaml "'yes'" # => "yes"
YAML::Any.from_yaml "yes" # => true
YAML.parse "'yes'" # => "yes"
YAML.parse "yes" # => true
```

Reply:
Please do add the necessary specs

Reply:
> Ah `YAML.parse` and `YAML::Any.from_yaml` seem to go through different codepaths?

Yeah, that's all a bit weird and hard to follow through, what's actually going on. YAML parsing could probably use some refactoring just to get the code clarified.

Reply:
Tests added :)